tensorflow/tools/dist_test/python/mnist_replica.py

```diff
@@ -94,68 +89,38 @@
 WORKER_PREFIX = "tf-worker"  # Prefix of the workers' domain names
 
 
-def get_cluster_setter():
-  """Get a cluster setter.
-
-  Returns:
-    Cluster setter and server object.
-  """
+def main(unused_argv):
+  mnist = input_data.read_data_sets(FLAGS.data_dir, one_hot=True)
+  if FLAGS.download_only:
+    sys.exit(0)
 
+  print("job name = %s" % FLAGS.job_name)
+  print("task index = %d" % FLAGS.task_index)
+  
+  #Construct the cluster and start the server
   ps_spec = FLAGS.ps_hosts.split(",")
   worker_spec = FLAGS.worker_hosts.split(",")
 
-  print(ps_spec)
-  print(worker_spec)
-
   cluster = tf.train.ClusterSpec({
       "ps": ps_spec,
       "worker": worker_spec})
   server = tf.train.Server(cluster,
                            job_name=FLAGS.job_name,
-                           task_index=FLAGS.worker_index)
-
-  # Get device setter from the cluster spec
-  return cluster, server
-
-
-def main(unused_argv):
-  mnist = input_data.read_data_sets(FLAGS.data_dir, one_hot=True)
-  if FLAGS.download_only:
-    sys.exit(0)
-
-  print("Worker GRPC URL: %s" % FLAGS.worker_grpc_url)
-  print("Worker index = %d" % FLAGS.worker_index)
-  print("Number of workers = %d" % FLAGS.num_workers)
-
-  # Sanity check on the number of workers and the worker index
-  if FLAGS.worker_index >= FLAGS.num_workers:
-    raise ValueError("Worker index %d exceeds number of workers %d " %
-                     (FLAGS.worker_index, FLAGS.num_workers))
-
-  # Sanity check on the number of parameter servers
-  if FLAGS.num_parameter_servers <= 0:
-    raise ValueError("Invalid num_parameter_servers value: %d" %
-                     FLAGS.num_parameter_servers)
-
-  # Construct cluster setter object
-  cluster, server = get_cluster_setter()
+                           task_index=FLAGS.task_index)
   
   if FLAGS.job_name == "ps":
     server.join()
   elif FLAGS.job_name == "worker":
```

Comment:
Add an `else:` block to this statement that raises an exception if `FLAGS.job_name` is not set, or unknown.